src/types/EmbeddedPaymentElement.tsx

```diff
@@ -261,29 +258,41 @@ async function createEmbeddedPaymentElement(
   return new EmbeddedPaymentElement();
 }
 
-function rowSelectionConfigurationIsValid(
+function getConfigurationValidationError(
   configuration: EmbeddedPaymentElementConfiguration
-): boolean {
-  // Confgiruation is invalid if:
+): string | null {
+  // Confgiruation is invalid if either:
   // 1. Row selection behavior is immediateAction AND
   // 2. Form sheet action is confirm AND
   // 3. Either Apple Pay is enabled OR customer configuration is present
+  // OR
+  // 1. Row selection behavior is NOT immediateAction AND
+  // 2. Row style is flatWithChevron
 
   const isImmediateAction =
     configuration.rowSelectionBehavior?.type === 'immediateAction';
   const isFormSheetConfirm = configuration.formSheetAction?.type === 'confirm';
   const isApplePayEnabled = !!configuration.applePay;
   const hasCustomerConfig = !!configuration.customerId;
+  const isFlatWithChevron =
+    configuration.appearance?.embeddedPaymentElement?.row?.style ===
+    'flatWithChevron';
 
+  // First condition: immediateAction + confirm + (ApplePay OR customer)
   if (
     isImmediateAction &&
     isFormSheetConfirm &&
     (isApplePayEnabled || hasCustomerConfig)
   ) {
-    return false;
+    return "Using 'immediateAction' with 'confirm' form sheet action is not supported when Apple Pay or a customer configuration is provided. Use 'default' row selection behavior or disable Apple Pay and saved payment methods.";
+  }
+
+  // Second condition: NOT immediateAction + flatWithChevron
+  if (!isImmediateAction && isFlatWithChevron) {
+    return "The 'flatWithChevron' row style can only be used with 'immediateAction' row selection behavior. Set rowSelectionBehavior to 'immediateAction' or use a different row style.";
```

Comment:
Same w/ my other comment in the other PR. Would be nice to not have to duplicate this logic if we don't have to since both native SDKs already do this validation.